Sources/Sentry/SentryHttpTransport.m

```diff
@@ -356,14 +357,21 @@ - (void)sendAllCachedEnvelopes
     // We must set sentAt as close as possible to the transmission of the envelope to Sentry.
     rateLimitedEnvelope.header.sentAt = [self.dateProvider date];
 
-    NSError *requestError = nil;
-    NSURLRequest *request = [self.requestBuilder createEnvelopeRequest:rateLimitedEnvelope
-                                                                   dsn:self.options.parsedDsn
-                                                      didFailWithError:&requestError];
+    if (nil == self.options.parsedDsn) {
+        SENTRY_LOG_FATAL(@"No DSN configured, not sending envelope.");
+        [self deleteEnvelopeAndSendNext:envelopeFilePath];
```

Comment:
`h`: I'm unsure if this change makes sense. If we don't have a valid parsed DSN, this will start deleting the current envelope without sending it, which is new behaviour. I think we should log an error or a warning and return here. Then the logic stays the same.